### PR TITLE
Andrei/agora 2296 endorsed delegates

### PR DESCRIPTION
**Pull request by andreitr**
- Implemented endorsed badge 
- Implemented endorsed filter
- Added endorsed tenant toggle
- Removed a filter to display delegates even with 0 votes

Screenshot 2024-07-18 at 2 47 23 PM

Screenshot 2024-07-18 at 2 48 11 PM

### CODE REVIEW

File: src/app/delegates/components/EndorsedFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      sort: "all",
+    },
+    true: {
+      value: "Endorsed Delegates",
+      sort: "true",
+    },
+  };
+
+  const handleChange = (value: string) => {
+    setIsDelegatesFiltering(true);
```
Comment (andreitr) [inline src/app/delegates/components/EndorsedFilter.tsx:33]: @mcgingras this flag is used to display the "flashing" loading state while the list is being filtered. I will have to refactor this later.

File: src/app/staking/components/delegates/DelegateCardList.tsx [hunk 2]
```diff
@@ -76,6 +76,7 @@ export default function DelegateCardList({
       >
         <DelegateCard
           key={address}
+          endorsed={false}
           address={address}
```
Comment (andreitr) [inline src/app/staking/components/delegates/DelegateCardList.tsx:80]: This is a default staking card - no need to display the endorsed checkmark.

File: src/app/api/common/delegates/getDelegates.ts [hunk 3]
```diff
@@ -242,25 +248,27 @@ async function getDelegates({
...
                 FROM agora.delegate_statements s
                 WHERE s.address = d.delegate
-                ${topIssuesQuery}
-                ${topStakeholdersQuery}
+                ${endorsedFilterQuery}
+                ${topIssuesFilterQuery}
+                ${topStakeholdersFilterQuery}
             )
-            ORDER BY -log(random()) / voting_power
+             ORDER BY -log(random()) / NULLIF(voting_power, 0)
```
[the remark this answers is not in the document]
Comment (mcgingras) [reply src/app/api/common/delegates/getDelegates.ts:271]: For cyber -- it seems like we are bringing back wanting to show voters with 0 power btw

File: src/app/staking/components/delegates/DelegateCardList.tsx [hunk 4]
```diff
@@ -101,6 +103,7 @@ export default function DelegateCardList({
             return (
               <DelegateCard
```
Comment (mcgingras) [inline src/app/staking/components/delegates/DelegateCardList.tsx:104]: I'm looking at the DelegateCard component you editing at `src/components/Delegates/DelegateCard/DelegateCard.tsx` and it only has `delegate` as a prop, I'm confused why this one accepts so many more props... weird. If it works it works though :)

**Review by mcgingras — APPROVED** [review]

**Comment by mcgingras** [comment]
The commit history on this branch is pretty weird, there are like 30 commits of @andreitr Merge branch 'main' of ...

File: src/app/api/common/delegates/getDelegates.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (andreitr) [reply src/app/api/common/delegates/getDelegates.ts:271]: @mcgingras yes, I am showing votes with 0 voting power.

File: src/app/api/common/delegates/getDelegates.ts [hunk 5]
```diff
@@ -412,8 +424,8 @@ async function getDelegate(addressOrENSName: string): Promise<Delegate> {
     address,
     contracts.alligator?.address || "",
     slug,
-    contracts.governor.address.toLowerCase(),
-    contracts.token.address.toLowerCase()
+    contracts.governor.address,
+    contracts.token.address
   );
```
Comment (jefag) [inline src/app/api/common/delegates/getDelegates.ts:429]: Why are you removing the `toLowerCase`?

I think we have conflicting patterns between tenants here.

**Review by jefag — APPROVED** [review]

**Pull request merged**